app/controllers/hiring_staff/vacancies/copy_controller.rb

```diff
@@ -1,17 +1,35 @@
 class HiringStaff::Vacancies::CopyController < HiringStaff::Vacancies::ApplicationController
+  def new
+    vacancy = Vacancy.find(vacancy_id)
+    @school = school
+    @copy_form = CopyVacancyForm.new(vacancy: vacancy)
+  end
+
   def create
     vacancy = Vacancy.find(vacancy_id)
-    vacancy_copy = CopyVacancy.new(vacancy: vacancy).copy
-    if vacancy_copy.save
-      Auditor::Audit.new(vacancy_copy, 'vacancy.copy', current_session_id).log
-      redirect_to review_path(vacancy_copy)
+    copy_form = CopyVacancyForm.new(vacancy: vacancy).apply_changes(copy_form_params)
```

Comment:
Thinking more about this (sorry!) I wonder whether it should have a bang at the end, because it's a destructive action, and I'm not sure about chaining the action either. You could probably do something like this:

```ruby
copy_form = CopyVacancyForm.new(vacancy: vacancy)
vacancy = copy_form.apply_changes!(copy_form_params)

if vacancy.valid?
  new_vacancy = CopyVacancy.new(original: vacancy, new: vacancy).call
....
```

You'll then also have access to the user's changes when showing the errors if the vacancy is invalid